lxqt-session/src/lxqtmodman.cpp

```diff
@@ -345,6 +345,12 @@ void LXQtModuleManager::logout()
         if (p->state() != QProcess::NotRunning && !p->waitForFinished(2000))
         {
             qCWarning(SESSION) << QString("Module '%1' won't terminate ... killing.").arg(i.key());
+
+            // disconnect everything connected to the module signals.
+            // Needed to prevent running LXQtModuleManager::restartModules()
+            // on an already destroyed process and the subsequent Q_ASSERT(proc)
+            // failure/crash
+            p->disconnect();
```

Comment:
@palinek Sorry for the delay. I was recovering for a minor surgery. Will post a detailed explanation tomorrow.